CSLabs.Api/Models/ModuleModels/Module.cs

```diff
@@ -56,6 +59,26 @@ public async Task SetUserModuleIdIfExists(DefaultContext context, User user)
                 UserModuleId = userModule.Id;
         }
 
+        public void AddModuleTags(DefaultContext context)
+        {
+            foreach (var moduleTag in ModuleTags.Where(moduleTag => moduleTag.TagId == 0))
+            {
+                context.Add(moduleTag);
+                if (moduleTag.Tag.Id != 0)
+                    context.Entry(moduleTag.Tag).State = EntityState.Unchanged;
```

Comment:
Why are checking if `id != 0` after you filtered the list down to where the `id == 0`?